src/native/crt.c

```diff
@@ -233,6 +233,38 @@ static void s_jni_atexit(void) {
     }
 }
 
+#define DEFAULT_MANAGED_SHUTDOWN_WAIT_IN_SECONDS 1
+
+static void s_jni_atexit_gentle(void) {
+
+    /* If not doing strict shutdown, wait only a short time before shutting down */
+    aws_thread_set_managed_join_timeout_ns(
+        aws_timestamp_convert(DEFAULT_MANAGED_SHUTDOWN_WAIT_IN_SECONDS, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL));
```

Comment:
we're going to add a 1 second pause on the shutdown of any application that forgets to close a handle?

what if we just **didn't even try to clean up anything**?

Reply:
We could do that too, but it makes me sad.

Reply:
I don’t think a 1sec pause is cool. like imagine someone wrote a tiny helper CLI type application that gets run in a loop by a larger script. This would make it painfully slow.

Let's either wait some infinitesimally short time, or not even try to clean up.

Reply:
chatted offline. there are advantages to at least trying to clean up.

we can always tweak this behavior going forward...